### DISCUSSION TITLE
[feature_request] Add total user stargazers object to GraphQL API

### DISCUSSION THREAD

**Discussion by rickstaa** [community/community#35677] [Apps, API and Webhooks]
Currently, there is no easy way to fetch the total stars a specific user has received on his/her repositories using the GraphQL API. Currently, this information is only available under the [repositoryconnection](https://docs.github.com/en/graphql/reference/objects#repositoryconnection) object. Providing users with a way to retrieve their total number of starGazers without looping through all repositories would improve the GraphQL API.

# Current behaviour

**Current graphQL query:**

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    login
    repositories(first: 100, ownerAffiliations: OWNER, orderBy: {direction: DESC, field: STARGAZERS}) {
      nodes {
        name
        stargazers {
          totalCount
        }
        }
        pageInfo {
          hasNextPage
          endCursor
        }
      }
  }
}
```

**Current GraphQL variables**

```graphql
{"login": "rickstaa"}
```

This will return the stargazers per repository.

# New behaviour

Making the `stargazers` objects available under the user level would improve the behaviour.

**New graphQL query:**

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    name
    login
    stargazers {
      totalCount
    }
  }
}
```

**New GraphQL variables**

```graphql
{"login": "rickstaa"}
```

### Using github pagination

I currently use [GraphQL pagination](https://graphql.org/learn/pagination/) to loop through all [repositories](https://docs.github.com/en/graphql/reference/objects#repositoryconnection). This, however, will take a long time.

**Comment by Riduidel** [comment]
Isn't it already implemented ?

When I do the following query

```
query {
  repository(owner: "django", name: "django") {
    stargazers(first: 100, after:"") {
      pageInfo {
        endCursor
        hasNextPage
        hasPreviousPage
        startCursor
      }
      edges {
        starredAt
        node {
          login
        }
      }
    }
    stargazerCount
  }
}

```
I have my `stargazerCount` available in response JSON

```
{
  "data": {
    "repository": {
      "stargazers": {
        "pageInfo": {
          "endCursor": "...",
          "hasNextPage": true,
          "hasPreviousPage": false,
          "startCursor": "..."
        },
        "edges": [
...
        ]
      },
      "stargazerCount": 79939
    }
  }
}
```
